api/types/role.go

```diff
@@ -952,6 +975,17 @@ func (r *RoleV6) SetGroupLabels(rct RoleConditionType, labels Labels) {
 	}
 }
 
+// CheckAndSetDefaults checks validity of all fields and sets defaults
+func (c *SPIFFERoleCondition) CheckAndSetDefaults() error {
+	if c.Path == "" {
+		return trace.BadParameter("path: should be non-empty")
+	}
+	if !strings.HasPrefix(c.Path, "/") {
+		return trace.BadParameter("path: should start with /")
+	}
```

Comment:
I mulled over this as I wrote this and from what I can tell, these are meaningless within the context of SPIFFE IDs - adding additional validation here would actually lead us to diverge from the SPIFFE specification. It would be entirely legitimate (although probably inadvisable) for an organisation to decide they wish to use `./` or `../` as part of their SPIFFE ID schema.